02-microarray/pathway-analysis_microarray_03_gsva.Rmd

```diff
@@ -364,48 +367,42 @@ max_dup_df <- mapped_df %>%
 
 max_dup_df
 ```
-Looks like we were able to successfully get rid of the duplicate gene identifiers and keep the observations with the maximum expression value!
+Looks like we were able to successfully collapse the duplicate gene identifiers by keeping the maximum expression value per sample associated with each of the duplicate gene identifiers!
 
 Now let's combine the rest of the mapped data!
 
 ```{r}
 filtered_mapped_df <- mapped_df %>%
-  dplyr::filter(
-    # First let's get the data associated with the gene identifiers that aren't duplicated
-    !entrez_id %in% dup_entrez_ids |
-      # Now let's get the data associated with the duplicated gene identifiers that we
-      # collapsed in `max_dup_df`
-      entrez_id %in% max_dup_df$entrez_id &
-        Ensembl %in% max_dup_df$Ensembl
-  )
+  # First let's get the data associated with the gene identifiers that aren't duplicated
+  dplyr::filter(!entrez_id %in% dup_entrez_ids) %>%
+  # Now let's bind the rows of the data we collapsed in `max_dup_df`
+  dplyr::bind_rows(max_dup_df) %>%
+  # Arrange so that the Entrez IDs that were duplicated are at the top
+  dplyr::arrange(dplyr::desc(entrez_id %in% dup_entrez_ids)) %>%
+  # Now let's remove the column we no longer need
+  dplyr::select(-"Ensembl")
 ```
 
-Let's check to see that we removed the duplicate Entrez IDs from the rest of the mapped data.
+Let's preview `filtered_mapped_df` to check that we combined the collapsed duplicate Entrez IDs data with the rest of the mapped data.
 
 ```{r}
-filtered_mapped_df %>%
-  # Let's sort by the Entrez IDs we know were duplicated
-  dplyr::arrange(dplyr::desc(entrez_id %in% dup_entrez_ids)) %>%
-  # Now lt's display the first six observations of our new object
-  head()
+head(filtered_mapped_df)
```

Comment:
If you are arranging early so the dups are at the top, we can leave the original `filtered_mapped_df`alone, but then here you can use `filtered_mapped_df %>% dplyr::filter(entrez_id %in% dup_entrez_ids)` to get the same effect.

Reply:
Gotcha 👍